packages/bitcore-wallet-client/src/lib/api.ts

```diff
@@ -671,26 +672,25 @@ export class API extends EventEmitter {
 
   _addSignaturesToBitcoreTx(txp, t, signatures, xpub) {
     const chain = Utils.getChain(txp.coin);
-    switch (chain) {
-      case 'ETH':
-        const unsignedTxs = t.uncheckedSerialize();
-        const signedTxs = [];
-        for (let index = 0; index < signatures.length; index++) {
-          const signed = CWC.Transactions.applySignature({
-            chain,
-            tx: unsignedTxs[index],
-            signature: signatures[index]
-          });
-          signedTxs.push(signed);
+    const network = txp.network;
+    if (!Constants.UTXO_COINS.includes(chain.toLowerCase())) {
```

Comment:
keeping the switch and just adding `case 'XRP':` before `case 'ETH'` could give us more flexibility for future coins.

Reply:
Changed back to switch statement